scripts/rosdistro_build_cache

```diff
@@ -86,13 +86,19 @@ def main():
         if args.dist_names and dist_name not in args.dist_names:
             continue
         # write the cache
-        data = yaml.dump(cache.get_data())
+        data = yaml.dump(cache.get_data(), Dumper=CacheYamlDumper)
+
         with open('%s-cache.yaml' % dist_name, 'w') as f:
             print('- write cache file "%s-cache.yaml"' % dist_name)
             f.write(data)
         with gzip.open('%s-cache.yaml.gz' % dist_name, 'wb') as f:
             print('- write compressed cache file "%s-cache.yaml.gz"' % dist_name)
-            f.write(data.encode('utf-8'))
+            try:
+                # Python 2 gzip module requires encoded data.
+                unicode
+                f.write(data)
+            except NameError:
+                f.write(data.encode('utf-8'))
```

Comment:
I can confirm that when always encoding, Python 2 craps out with:

```
Traceback (most recent call last):
  File "/usr/local/bin/rosdistro_build_cache", line 6, in <module>
    exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/Users/mikepurvis/rosdistro_py/scripts/rosdistro_build_cache", line 99, in <module>
    main()
  File "/Users/mikepurvis/rosdistro_py/scripts/rosdistro_build_cache", line 96, in main
    f.write(data.encode('utf-8'))
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 132821: ordinal not in range(128)
```

Your final suggestion is reasonable; I shall change it to that.